core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -19,11 +19,38 @@
 
 package org.apache.iceberg;
 
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
 public class TableProperties {
 
   private TableProperties() {
   }
 
+  /**
+   * Reserved table property for table format version.
+   * <p>
+   * Iceberg will default a new table's format version to the latest stable and recommended version.
+   * This reserved property keyword allows users to override the Iceberg format version of the table metadata.
+   * <p>
+   * If this table property exists when creating a table, the table will use the specified format version.
+   * If a table updates this property, it will try to upgrade to the specified format version.
+   * This helps developers to try out new features in an unreleased version or migrate existing tables to a new version.
+   * <p>
+   * Note: incomplete or unstable versions cannot be selected using this property.
```

Comment:
I don't think that this is actually enforced. Should we add a test and validate the versions that come from table properties?

Reply:
This is currently enforced because `SUPPORTED_TABLE_FORMAT_VERSION` is 2 and there are no unsupported versions. When we introduce v3, I think we should add a flag to `upgradeToFormatVersion` that controls whether to do the `SUPPORTED_TABLE_FORMAT_VERSION` check. That way the existing check validates this and there is a way in the future to "manually" update the format version to unsupported versions for testing.